docs/root/version_history/current.rst

```diff
@@ -70,6 +70,7 @@ New Features
 * config: added ability to flush stats when the admin's :ref:`/stats endpoint <operations_admin_interface_stats>` is hit instead of on a timer via :ref:`stats_flush_on_admin <envoy_v3_api_field_config.bootstrap.v3.Bootstrap.stats_flush_on_admin>`.
 * config: added new runtime feature `envoy.features.enable_all_deprecated_features` that allows the use of all deprecated features.
 * crash support: added the ability to dump L4 connection data on crash.
+* ext_authz: added :ref:`response_headers_to_add <envoy_v3_api_field_service.auth.v3.OkHttpResponse.response_headers_to_add>` to support sending response headers to downstream clients on OK external authorization checks.
```

Comment:
While is in the same spirit, seems like we have two changes here, for gRPC and HTTP implementation. Should I'm OK to have this in the same PR. For completeness, we need integration tests, since it involves response from the "external service". WDYT?

Reply:
I'll add another changelog line to capture both the HTTP and gRPC service changes.

I'll look into what it'll take to add an integration test  👍

Reply:
I took a look and there seems to be an established precedent that other similar features (e.g. allowed_client_headers for HTTP, request_headers_to_add for gRPC, and others) are tested using unit tests and appropriate mocks. I think that if unit tests are OK for those features, they should be good for this one as well. Do you agree?

Having said that, I don't think there are any integration tests for ext_authz (or if there are, I can't find them) so maybe I should look into following up with some work to add those tests down the road. Maybe I'll look at the coverage report to decide if that's worth it.

Reply:
I think we need to have a test to make sure that, for gRPC implementation, when we do `sendExtAuthzResponse` (this is done by the authorization service) with this `response_headers_to_add` set, the downstream client receives it. 

The same applies to the HTTP one, `waitForExtAuthzRequest` should provide response header entries that are allowed to be forwarded, and are inserted to `response_->headers()`.

Reply:
Ok I'll give that a try

Reply:
I now see the integration test for ext_authz - don't know how I missed that before.